eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -1,7 +1,33 @@
 jobs:
+  - job: Prepare
```

Comment:
By doing this in its own job you are getting pretty close to the original plan of just using one pipeline that does filtering up front :)

Reply:
But in all seriousness why did you put this in its own job as opposed to just a pre-step in the other jobs.

Reply:
Because each of the jobs is on different platforms I ran into subtle pathing issues with either bash or cmd.exe - the beauty of just having a single job that does these stage wide calculations is that you only have to script for that one platform and then carry the state across in variables.

Initially I did it in the Build job, but that would have meant that I needed to make the test runs depend on the build step. Given that this Prepare step takes about 25 seconds, I figured it would be optimal in terms of allowing for greater concurrency for the build and test legs.

Regarding your note about having a single pipeline that does filtering up front - yeah this is probably one of the ways we would have passed state between jobs in that scenario too, but this doesn't do anything around analyzing commit history to determine what paths should be searched :)